### PR TITLE
Add module AccountManagement

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Add new module Microsoft.PowerShell.AccountManagement
There is only single cmdlet New-User to keep the PR as small as possible.
Later we will add other cmdlets.

## PR Context

Related #2996

Module Microsoft.PowerShell.LocalAccounts was removed from PowerShell Core because of using non-public APIs.
Microsoft.PowerShell.AccountManagement uses only public APIs and based on System.DirectoryServices.AccountManagement.

It is impossible to get full backward compatibility with Microsoft.PowerShell.LocalAccounts module due to differences of the underlying API.
So the main model was chosen Microsoft ActiveDirectory module.
It will be easier for users to migrate from Microsoft.PowerShell.LocalAccounts to a more functional Microsoft.PowerShell.LocalAccounts module if necessary.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: assets/files.wxs
Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Test-Connection.Error occurred in Describe block
```powershell
Exception calling "GetHostEntry" with "1" argument(s): "Name or service not known"
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:24 char:9
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 3 failures in `PowerShell-CI-windows`
Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Cannot process command because of one or more missing mandatory parameters: Name Password.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 40
40:             $result = New-User -AccountStore Machine

```
Validate New-User.Errors on Name argument of empty string, null, spaces
```powershell
Expected an exception, with FullyQualifiedErrorId 'InvalidValue,Microsoft.PowerShell.Commands.NewUserCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 107
107:             { New-User -Name "   " -PasswordNotRequired } | Should -Throw -ErrorId "InvalidValue,Microsoft.PowerShell.Commands.NewUserCommand"
```
Validate New-User.Error on user exists
```powershell
Expected an exception, with FullyQualifiedErrorId 'ParameterArgumentValidationError,Microsoft.PowerShell.Commands.NewUserCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 112
112:             $exc = { New-User -Name TestUserNew -PasswordNotRequired -AccountStore Machine } | Should -PassThru -Throw -ErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.NewUserCommand"
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 2 failures in `PowerShell-CI-windows`
Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Parameter set cannot be resolved using the specified named parameters. One or more parameters issued cannot be used together or an insufficient number of parameters were provided.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 40
40:             $result = New-User -AccountStore Machine `

```
Validate New-User.Error on user exists
```powershell
Expected an exception, with FullyQualifiedErrorId 'ParameterArgumentValidationError,Microsoft.PowerShell.Commands.NewUserCommand' to be thrown, but the FullyQualifiedErrorId was 'InvalidValue,Microsoft.PowerShell.Commands.NewUserCommand'. from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:112 char:22
    + ...    $exc = { New-User -Name TestUserNew -PasswordNotRequired -AccountS ...
    +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 112
112:             $exc = { New-User -Name TestUserNew -PasswordNotRequired -AccountStore Machine -ErrorAction Stop } | Should -PassThru -Throw -ErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.NewUserCommand"
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 2 failures in `PowerShell-CI-windows`
Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected strings to be the same, but they were different.
Expected length: 11
Actual length:   5
Strings differ at index 0.
Expected: 'TestUserNew'
But was:  'samq2'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 51
51:             $result.Name | Should -BeExactly TestUserNew
```
Validate New-User.Error on user exists
```powershell
A parameter cannot be found that matches parameter name 'BeTypeOf'.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 113
113:             $exc.Exception | Should -BeTypeOf "System.DirectoryServices.AccountManagement.PrincipalExistsException"

```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.Ping.Delay works
```powershell
Expected the actual value to be less than 3, but got 3.3178178999999997.
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 150
150:             $result1.TotalSeconds | Should -BeLessThan 3
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected $true, but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 54
54:             $result.AccountNeverExpires | Should -BeTrue
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected exactly 'sn', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 58
58:             $result.SurName | Should -BeExactly "sn"
```

**Comment by TravisEz13** [comment]
is the syntax the same as the windows PowerShell cmdlet?

**Comment by iSazonov** [comment]
@TravisEz13 No, as mentioned in the PR description the used API is another and it impossible to keep _full_ backward compatibility. The new design is closer to ActiveDirectory module. It make more sense because scripts can works (manage accounts) without the need to install AdminTools everyware. 
I did not look at this in detail, but it is quite possible that we could get almost the same functionality using the cmdlet aliases and parameters. I suppose it should stay outside of this module.

**Comment by iSazonov** [comment]
Last days CI Macos frequently time out.

File: assets/files.wxs
Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 60
60:             $result.Enabled | Should -BeTrue
```

**Comment by iSazonov** [comment]
@SteveL-MSFT  Can your team review so that we can move on and add more cmdlets in follow PRs?

**Comment by kilasuit** [comment]
IMO this should be externalised, released to the Gallery and pulled into the output build of PowerShell (like PowerShellGet) instead of being added to the codebase

**Comment by adityapatwardhan** [comment]
I agree with @kilasuit. This should ideally go to PSGallery and be pulled down.

**Comment by iSazonov** [comment]
We have a base account management in Windows PowerShell many years. If plan is to replace Windows PowerShell with PowerShell 7.0 we need to get back most of common modules and features that we lost in porting time. 

- Please point me the community that is ready to take on the production of this module for all the necessary platforms. Over the past 3 years, I have repeatedly heard that we need to decouple modules. But where are the people who take it upon themselves? I am ready to make the module (and make other contributions) _in the repo_ but I am not ready to take on the entire production cycle.

- Currently I have 99.9% needed PowerShell features in-box on all my enterprise servers to manage the infrastructure. Today we (I say about enterprise) cannot replace Windows PowerShell with PowerShell Core because it is slow, consumes a lot of resources and, mostly, lost a lot of functionality. (I do not think that it is possible to fix it a few months before the 7.0 release.) We can not freely use third-party modules because strong enterprise policy - we can not trust whole Internet, validate every new version, we have isolated networks. PowerShell for us is an enterprise level management tool. If it will have not all we had many years we will have to spend time and money to implement/find workarounds (maybe write internal scripts to replace gaps or more likely to replace it with a better alternative management tool).

**Comment by vexx32** [comment]
I don't think there's any necessary need to force these to be a completely separate module.

Maybe have a tracking issue or issues for such modules that we want to eventually have out of box? Separating them to a separately shipped module and/or pulling them into the PS build cycle isn't a huge undertaking in and of itself.

I don't see why we need reject this as it is at the present; we can always move to separate it to a separate module when someone willing to maintain and develop it further steps up, whether from the PS team or the community.

**Comment by SteveL-MSFT** [comment]
I think we would want to include this as part of the PS7 package, but also think it should be a separate module that is published to PSGallery and have its own repo.  Eventually, I want to split up and move Utility and Management modules out of this repo (see https://github.com/PowerShell/PowerShell/issues/9960).  Since the code is all self contained in a single folder, I think it's ok to just keep it in this repo for now and we can move it out when we do so for the others.  By that time, hopefully we'll have templatized how to create the repos, contributing docs/etc.., enable CI, etc...

File: src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs]: Null here means never expires?  But you have `-AccountNeverExpires` below?

File: src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs]: Does it make sense to be Domain by default?  Does this work on non-Windows?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by SteveL-MSFT** [comment]
Should have the @PowerShell/powershell-committee do a cmdlet review

File: src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs
Comment (iSazonov) [reply src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs]: It is only Windows API. Although it is based on "providers" and we could get support Unix PAM or something like in future.

File: src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs
Comment (iSazonov) [reply src/Microsoft.PowerShell.AccountManagement/commands/NewUserCommand.cs]: It doesn't work. Will remove.

File: src/Modules/Windows/Microsoft.PowerShell.AccountManagement/Microsoft.PowerShell.AccountManagement.psd1
Comment (iSazonov) [reply src/Modules/Windows/Microsoft.PowerShell.AccountManagement/Microsoft.PowerShell.AccountManagement.psd1]: Fixed.

File: src/Modules/Windows/Microsoft.PowerShell.AccountManagement/Microsoft.PowerShell.AccountManagement.psd1
Comment (iSazonov) [reply src/Modules/Windows/Microsoft.PowerShell.AccountManagement/Microsoft.PowerShell.AccountManagement.psd1]: Fixed.

**Comment by iSazonov** [comment]
My main concern about module decouples is that we will waste a lot of time and resources without benefits.


Let's look at PowerShell-Native lib. How many benefits did we get? I think we only complicate dev process. Before the decouple I could make PR in the PowerShell repo and get test results in minutes. Now I do not even know how long it takes to set up the environment, compile and get test results (only locally!). This is not so important as the module is stable.

Main benefit from the decoupling I see for projects which is under high community development activity. 
Have we such modules in the repo? No, all modules is stable.
I guess we could develop new HelpSystem module in separate repo but first we should decouple current one from engine but after that I don't expect high community activity so the moving in a separate project loses its meaning - if someone wants to make another module based on new public API he can do it in himself repo.

Second benefit is the ability to create a PowerShell distributive of the minimum size.

We could easily resolve this like .Net Core does. They have a runtime with most of needed code and dozen packages.
This means that important frequently used modules live in this PowerShell repository, but are also published as packages (or/and directly included in distributive).

**Comment by SteveL-MSFT** [comment]
@iSazonov yes, we could still publish the modules in PSGallery while keeping the code in this repo.  It may depend on whether someone (or some people) step up to be maintainers of that module to justify putting it in separate repo.

**Comment by kilasuit** [comment]
quoting @SteveL-MSFT 
>I think we would want to include this as part of the PS7 package, but also think it should be a separate module that is published to PSGallery and have its own repo. Eventually, I want to split up and move Utility and Management modules out of this repo (see #9960). Since the code is all self contained in a single folder, I think it's ok to just keep it in this repo for now and we can move it out when we do so for the others. By that time, hopefully we'll have templatized how to create the repos, contributing docs/etc.., enable CI, etc...

In all honesty this is something that has needed prioritising for far too long now and with each month that passes with new additions like this suggested new module it makes the task harder and harder and IMO unlikely that it will ever happen. The community have tried to help get steering on this many times before however the issue becomes a case of it being very opinionated on how the best way forward is. I personally want to see this happen for v7 

My main concern about bundling more and more into this repo is unnecessary bloat that is not critical to the running and using of the core engine of PowerShell. PowerShellGet/PSReadline are 2 examples where they aren't engine critical so rightly live outside the engine and can be pulled in at packaging time.

Honestly as part of v7 plans you should minimize the amount of code being added to this repo that is new functionality that is not v7 critical (this change IMO is not critical for PowerShell use in v7) and therefore should be externalised and any new modules to be bundled into this repo should go through RFC Process IMO

Also see my comment in #9960 - this to me feels like the wrong place to put effort in. 


> We could easily resolve this like .Net Core does. They have a runtime with most of needed code and dozen packages.

This would be the output of this issue would it not - #5681 - which is where I would like to see v7 lead to. with default install being minimal but additional packages available to bootstrap other uses (which in windows could be additional check boxes in the additional features part of programs and features)

**Comment by joeyaiello** [comment]
@PowerShell/powershell-committee agrees that this module should have an RFC, irrespective of whether it's packaged with PS7 or released only to the Gallery. Currently, we're unclear on whether this would be cross-platform, as well as what the all-up cmdlet surface area would look like. (Even if you start with one cmdlet, it should make sense in the context of a planned design for the rest of the module.)

We also just commented in #9960, and agree that the decision there should not block whether we accept this PR. 

@iSazonov I totally hear your requirements around not being able to pull things in from the web, and to that end I think it's important to separate the two issues around shipping modules in the Gallery and where the source code ultimately lives. Those should be independent, and we should close them in #9960.

That being said, we do believe it makes sense for this to start as a Gallery module for validation. This isn't to say it should never be included in any PS7 package of any kind, but it would allow you to iterate much more quickly than if we are constrained by the RFC process. And in fact, you could publish to the Gallery today, we on the team and in the community could play with it, and your implementation can serve as a much easier basis for writing and reviewing an RFC about taking it into PS7. 

(And we discussed that it might be a good idea to actually make that the general guidance for contributing new modules all up.)

**Comment by iSazonov** [comment]
For full understanding, my main intention here was/is only to plug a hole with LocalAccount module before 7.0 release. Nothing more :-)

Looking back, I see how the MSFT contribution policy has changed from very strict to liberal. _Automation tools contributed to this._ Now, I even speculated that the number of contributors has increased - because the dev process has become easier to do PRs.

Now I dream that this process will move to a new level. :-)

Probably many of us have indoor flowers. They are not difficult to care for and water, and they bring pleasure. Fewer people are ready to plant flowers in front of the house because it takes more time and effort. And very few people are willing to go a long way to devote their time somewhere to caring for flowers, and even more so to do it in many places just for fun.
I mean - most members of the community are present here for fun and not business. The further you move the code away from the center and the more it is sprayed on the repositories, the more costs you cause and the less pleasure it brings.

The best model for community contributions that MSFT invented is the CoreFX / CoreCLR model. They separated auxiliary projects like performance, installation, core tools, etc. but combined all featured code in the center (note that this is a _huge_ number of packages).

I pulled PR with Simple Case Folding to corefxlab repo and ask after the PR merging when the package will be published - the answer was that it was already auto published immediately after the merging.

If PowerShell team wants to move forward with in integrating with dev community, it could create a similar infrastructure so that contributors can promote modules just as easily as in Core.

I am pleased to take care of the flowers in this room, but I am not ready to rush through wild remote plantations. :-)

File: assets/files.wxs
Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 10, but got 0.764912312888602.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1921
1921:             $pathologicalRatio | Should -BeGreaterThan 10
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.Error occurred in Describe block
```powershell
Exception calling "GetHostEntry" with "1" argument(s): "nodename nor servname provided, or not known"
At /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:15 char:9
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Validate AccountManagement user cmdlets.Validate Get-User.Error occurred in Context block
```powershell
Cannot index into a null array.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.AccountManagement/AccountManagement.User.Tests.ps1:163 char:17
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Validate AccountManagement user cmdlets.Validate Get-User.Error occurred in Context block
```powershell
Cannot index into a null array.
At /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.AccountManagement/AccountManagement.User.Tests.ps1:163 char:17
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 7 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with only name
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 27
27:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with password
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 38
38:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected exactly 'samq2', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 54
54:                 $result.Name | Should -BeExactly "samq2"
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with pipeline
```powershell
Expected an exception, with FullyQualifiedErrorId 'UserAlreadyRemoved,Microsoft.PowerShell.Commands.RemoveUserCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 137
137:                 { $userIdentity | Remove-User -ErrorAction Stop } | Should -Throw -ErrorId "UserAlreadyRemoved,Microsoft.PowerShell.Commands.RemoveUserCommand"
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with parameter
```powershell
Expected no exception to be thrown, but an exception "Cannot bind argument to parameter 'Identity' because it is null." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:141 char:41
    +                 { Remove-User -Identity $userIdentity -ErrorAction St ?
    +                                         ~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 141
141:                 { Remove-User -Identity $userIdentity -ErrorAction Stop } | Should -Not -Throw
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 7 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with only name
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 27
27:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with password
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 38
38:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected exactly 'samq2', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 54
54:                 $result.Name | Should -BeExactly "samq2"
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with pipeline
```powershell
Expected no exception to be thrown, but an exception "Cannot access a disposed object.
Object name: 'PrincipalContext'." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:134 char:35
    + ?               { $userIdentity | Remove-User -ErrorAction Stop } | Sho ?
    +                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 134
134:                 { $userIdentity | Remove-User -ErrorAction Stop } | Should -Not -Throw
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with parameter
```powershell
Expected no exception to be thrown, but an exception "Cannot bind argument to parameter 'Identity' because it is null." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:141 char:41
    +                 { Remove-User -Identity $userIdentity -ErrorAction St ?
    +                                         ~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 141
141:                 { Remove-User -Identity $userIdentity -ErrorAction Stop } | Should -Not -Throw
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 10 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with only name
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 27
27:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with password
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 38
38:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected exactly 'samq2', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 54
54:                 $result.Name | Should -BeExactly "samq2"
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with pipeline
```powershell
Expected no exception to be thrown, but an exception "Cannot access a disposed object.
Object name: 'PrincipalContext'." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:134 char:35
    + ?               { $userIdentity | Remove-User -ErrorAction Stop } | Sho ?
    +                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 134
134:                 { $userIdentity | Remove-User -ErrorAction Stop } | Should -Not -Throw
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with parameter
```powershell
Expected no exception to be thrown, but an exception "Cannot bind argument to parameter 'Identity' because it is null." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:141 char:41
    +                 { Remove-User -Identity $userIdentity -ErrorAction St ?
    +                                         ~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 141
141:                 { Remove-User -Identity $userIdentity -ErrorAction Stop } | Should -Not -Throw
```

Comment (PoshChan) [inline assets/files.wxs]: @iSazonov, your last commit had 7 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with only name
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 27
27:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with password
```powershell
Expected exactly 'TestUserNew', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 38
38:                 $result.Name | Should -BeExactly TestUserNew
```
Validate AccountManagement user cmdlets.Validate New-User.Can create New-User with explicit properties in local machine
```powershell
Expected exactly 'samq2', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 54
54:                 $result.Name | Should -BeExactly "samq2"
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with pipeline
```powershell
Expected no exception to be thrown, but an exception "Cannot access a disposed object.
Object name: 'PrincipalContext'." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:134 char:35
    + ?               { $userIdentity | Remove-User -ErrorAction Stop } | Sho ?
    +                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 134
134:                 { $userIdentity | Remove-User -ErrorAction Stop } | Should -Not -Throw
```
Validate AccountManagement user cmdlets.Validate Remove-User.Can remove user account with parameter
```powershell
Expected no exception to be thrown, but an exception "Cannot bind argument to parameter 'Identity' because it is null." was thrown from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1:141 char:41
    +                 { Remove-User -Identity $userIdentity -ErrorAction St ?
    +                                         ~~~~~~~~~~~~~.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.AccountManagement\AccountManagement.User.Tests.ps1: line 141
141:                 { Remove-User -Identity $userIdentity -ErrorAction Stop } | Should -Not -Throw
```

**Comment by iSazonov** [comment]
Closed with https://github.com/PowerShell/PowerShell/pull/12242#issuecomment-637072659

**Pull request closed without merging**
